### PR TITLE
add current_page and current_page_backwards parameters to cursor based pagination

### LINKED ISSUE #744: No current page on cursor based pagination
I am building an app with bi-directional infinite scrolling, where the user can scroll up through pages of old posts, and new posts occasionally appear on the opposite direction. There is currently no way to refetch the first page (as it was fetched without a cursor, to see if there are new pages added on the front. There is also the reverse use case which would be refetching the last page to see if new pages have been added on the end.

`sqlakeyset` supports this with:

```python
page.paging.bookmark_current
page.paging.bookmark_current_backwards
```

Can this get added to the cursor paginated response?

### PR DESCRIPTION
**Pull request by cranium**
Resolves #744

### CODE REVIEW

File: fastapi_pagination/cursor.py [hunk 1]
```diff
@@ -75,6 +75,11 @@ def to_raw_params(self) -> CursorRawParams:
 class CursorPage(AbstractPage[T], Generic[T]):
     items: Sequence[T]
 
+    current_page: str = Field(..., description="Cursor to refetch the current page")
+    current_page_backwards: str = Field(
+        ...,
+        description="Cursor to refetch the current page starting from the last item",
+    )
```
Comment (uriyyo) [inline fastapi_pagination/cursor.py:82]: Let's make those fields optional cause they can't be unavailable for other extenstions

File: tests/ext/test_sqlalchemy_cursor.py [hunk 2]
```diff
@@ -31,6 +31,14 @@ def get_db() -> Iterator[Session]:
     def route(db: Session = Depends(get_db)):
         return paginate(db, select(sa_user).order_by(sa_user.id, sa_user.name))
 
+    @app.get("/first-85", response_model=CursorPage[UserOut])
+    def route(db: Session = Depends(get_db)):
+        return paginate(db, select(sa_user.where(sa_user.id <= 85)).order_by(sa_user.id, sa_user.name))
```
Comment (uriyyo) [inline tests/ext/test_sqlalchemy_cursor.py:36]:
```suggestion
        return paginate(db, select(sa_user).where(sa_user.id <= 85).order_by(sa_user.id, sa_user.name))
```

File: tests/ext/test_sqlalchemy_cursor.py [hunk 3]
```diff
@@ -31,6 +31,14 @@ def get_db() -> Iterator[Session]:
...
+
+    @app.get("/last-85", response_model=CursorPage[UserOut])
+    def route(db: Session = Depends(get_db)):
+        return paginate(db, select(sa_user.where(sa_user.id > 15)).order_by(sa_user.id, sa_user.name))
```
Comment (uriyyo) [inline tests/ext/test_sqlalchemy_cursor.py:40]:
```suggestion
        return paginate(db, select(sa_user).where(sa_user.id > 15).order_by(sa_user.id, sa_user.name))
```

**Review by uriyyo — CHANGES_REQUESTED** [review]

File: fastapi_pagination/cursor.py (hunk 1 above)
Comment (cranium) [reply fastapi_pagination/cursor.py]: Thanks for the feedback! Is this a future proofing thing? I didn't see anything but the sqlalchemy extension using this.

**Comment by cranium** [comment]
Thanks for the comments @uriyyo - let me know if you think of anything else!

File: fastapi_pagination/cursor.py [hunk 4]
```diff
@@ -86,13 +91,17 @@ def create(
         items: Sequence[T],
         params: AbstractParams,
         *,
+        current: Cursor,
+        current_backwards: Cursor,
```
Comment (uriyyo) [inline fastapi_pagination/cursor.py:95]: Make new params optional:
```suggestion
        current: Optional[Cursor] = None,
        current_backwards: Optional[Cursor] = None,
```

**Review by uriyyo — CHANGES_REQUESTED** [review]
LGTM, I have only one minor comment

File: fastapi_pagination/cursor.py (hunk 1 above)
Comment (uriyyo) [reply fastapi_pagination/cursor.py]: There are other extensions that use `CursorPage`:
https://github.com/uriyyo/fastapi-pagination/blob/main/fastapi_pagination/ext/cassandra.py

**Review by uriyyo — APPROVED** [review]
LGTM, thanks

**Comment by cranium** [comment]
Thanks @uriyyo! Stay safe my friend!

**Comment by uriyyo** [comment]
@cranium Thanks ❤️

**Pull request merged**